heartbeat/scheduler/scheduler.go

```diff
@@ -200,6 +216,20 @@ func (s *Scheduler) Add(sched Schedule, id string, entrypoint TaskFunc, jobType
 		}
 		debugf("Job '%v' returned at %v", id, time.Now())
 	}
+	isActive := false
+	for _, pmw := range pmws {
+		if pmw.IsActive(time.Now()) {
+			isActive = true
+			break
+		}
+	}
+
+	if s.runOnce && isActive {
+		return func() {
+			debugf("Remove scheduler job '%v'", id)
+			jobCtxCancel()
+		}, nil
+	}
```

Comment:
I don't think this section is needed here, run once monitors are going through `taskFn` too so it's duplicated

Reply:
i have removed it. i will test it manually now

Reply:
i have tested and you are right it works for both.